### DISCUSSION TITLE
.csv device upload failing with"unexpected header" for the first column

### DISCUSSION THREAD

**Discussion by NateMillerTR** [netbox-community/netbox#8044] [Q&A] [answered]
Netbox 3.0.4

I've generated a .csv file through excel in osx-  I save a sheet as a .csv
If I click Devices -> upload ->Racked Devices -> CSV File Upload

I get the error "!CSV file - Unexpected column header "name" found"

The example starts with "device_role", so I moved the name column, but now the problem is "unexpected column header "device_role" found"

If I look at the .csv, the first line is exactly this:
`name,device_role,tenant,manufacturer,device_type,platform,serial,asset_tag,status,site,location,rack,position,face,comments`

If I take the contents of the .csv and copy and paste them into the CSV Data window, the bulk upload works.
So it doesn't appear to be the data per se, but I don't know what's going wrong with the data.

Does the .csv file header line need a different format for the headers?

**Comment by candlerb** [comment, marked as answer]
Since it's the *first* column that has the problem, I suspect that Excel has put some hidden magic character at the start of the file.  Perhaps it's a unicode BOM. If this turns out to be the case, then you can open a feature request for the CSV upload to process (or ignore) the BOM.

Try printing the first few bytes of the file like this:

```
head -c32 myfile.csv | hexdump -C
```

and you can paste it here if it doesn't make sense.

**Reply by candlerb** [reply]
I just tested this with Excel for Mac:

```
$ hexdump -C ~/Documents/blah.csv
00000000  ef bb bf 61 73 64 66 2c  67 68 6a 6b 2c 71 65 77  |...asdf,ghjk,qew|
00000010  72 0d 0a 70 6f 69 75 2c  69 75 79 74 2c 75 79 74  |r..poiu,iuyt,uyt|
00000020  72                                                |r|
00000021
```

`ef bb bf` is indeed the UTF-8 [BOM](https://en.wikipedia.org/wiki/Byte_order_mark).  You can strip the first three bytes of the file like this:

```
tail -c +4 foo.csv >foo2.csv
```

However I think there's a simpler solution: just choose the non-UTF-8 version of CSV when exporting, as long as your data is plain ASCII without any special symbols like accented characters.

![image](https://user-images.githubusercontent.com/44789/145707638-1457873c-8f74-42dd-9ef5-22e9817b7975.png)

**Reply by NateMillerTR** [reply]
You were spot on, thank you very much.  I had thought that "scrubbing" it through textedit would have revealed/removed any odd fingerprints like that.

It'd be nice if the file import files procedure had a basic sanitizer-  check for a BOM, account for various line feed/new line combinations, etc.  

I appreciate your help!
